src/directLine.ts

```diff
@@ -869,4 +885,40 @@ export class DirectLine implements IBotConnection {
             )
         )
     }
+
+    private commonHeaders() {
+
+
+        if (this._userAgent === '') {
+            try {
+                this._userAgent = window.navigator.userAgent || '';
+            } catch {
+                try {
+                    // set node user agent
+                    // @ts-ignore
+                    const os = require('os');
+                    const { arch, platform, version } = process;
+                    this._userAgent = `Node.js,Version=${version}; ${platform} ${os.release()}; ${arch}`
+                } catch {
+                    // no-op
+                }
+            }
+        }
+
+        return {
+            "Authorization": `Bearer ${this.token}`,
+            "User-Agent": `${this._botAgent} (${this._userAgent})`,
+            "x-ms-bot-agent": this._botAgent
+        };
+    }
+
+    private getBotAgent(customAgent: string = ''): string {
+        let clientAgent = `directlinejs/${process.env.VERSION || '0.0.0'}`
+
+        if (customAgent) {
+            clientAgent += `; ${customAgent}`
+        }
+
+        return `DirectLine/3.0 (${clientAgent})`;
```

Comment:
should DirectLine/3.0 be saved as a const somewhere? CURR_DIRECTLINE_VERSION or something

Reply:
Good call.